isisd/isis_zebra.c

```diff
@@ -225,11 +212,102 @@ void isis_zebra_route_add_route(struct isis *isis,
 		}
 
 		api_nh->ifindex = nexthop->ifindex;
+
+		/* Add MPLS label(s). */
+		switch (type) {
+		case ISIS_ROUTE_NEXTHOP_MAIN:
+		case ISIS_ROUTE_NEXTHOP_BACKUP:
+			/*
+			 * SR/TI-LFA labels are installed using separate
+			 * messages.
+			 */
+			break;
+		case ISIS_MPLS_NEXTHOP_MAIN:
+			if (nexthop->sr.label != MPLS_INVALID_LABEL) {
+				api_nh->label_num = 1;
+				api_nh->labels[0] = nexthop->sr.label;
+			} else {
+				api_nh->label_num = 1;
+				api_nh->labels[0] = MPLS_LABEL_IMPLICIT_NULL;
+			}
+			break;
+		case ISIS_MPLS_NEXTHOP_BACKUP:
+			if (nexthop->label_stack) {
+				api_nh->label_num =
+					nexthop->label_stack->num_labels;
+				memcpy(api_nh->labels,
+				       nexthop->label_stack->label,
+				       sizeof(mpls_label_t)
+					       * api_nh->label_num);
+			} else {
+				api_nh->label_num = 1;
+				api_nh->labels[0] = MPLS_LABEL_IMPLICIT_NULL;
+			}
+			break;
+		}
+
+		/* Backup nexthop handling. */
+		if (backup_nhs) {
+			SET_FLAG(api_nh->flags, ZAPI_NEXTHOP_FLAG_HAS_BACKUP);
+			/*
+			 * If the backup has multiple nexthops, all of them
+			 * protect the same primary nexthop since ECMP routes
+			 * have no backups.
+			 */
+			api_nh->backup_num = backup_nhs;
+			for (int i = 0; i < backup_nhs; i++)
+				api_nh->backup_idx[i] = i;
+		}
 		count++;
 	}
-	if (!count)
+
+	return count;
+}
+
+void isis_zebra_route_add_route(struct isis *isis, struct prefix *prefix,
+				struct prefix_ipv6 *src_p,
+				struct isis_route_info *route_info)
+{
+	struct zapi_route api;
+	int count = 0;
+
+	if (zclient->sock < 0)
 		return;
 
+	memset(&api, 0, sizeof(api));
+	api.vrf_id = isis->vrf_id;
+	api.type = PROTO_TYPE;
+	api.safi = SAFI_UNICAST;
+	api.prefix = *prefix;
+	if (src_p && src_p->prefixlen) {
+		api.src_prefix = *src_p;
+		SET_FLAG(api.message, ZAPI_MESSAGE_SRCPFX);
+	}
+	SET_FLAG(api.message, ZAPI_MESSAGE_NEXTHOP);
+	SET_FLAG(api.message, ZAPI_MESSAGE_METRIC);
+	api.metric = route_info->cost;
+#if 0
+	SET_FLAG(api.message, ZAPI_MESSAGE_DISTANCE);
+	api.distance = route_info->depth;
+#endif
```

Comment:
Please remove dead code

Reply:
Done. I tried to preserve that `#if 0` block from the original code but in fact setting the admin distance that way doesn't make any sense :)